### PR TITLE
Add extra fields to Person model

### PR DESCRIPTION
**Pull request by sdiv877**
Addresses #114.
* Replaces `first_name` and `last_name` in the `Person` model with `full_name`.
* Replaces `time_added` with `time_updated`
* Adds `gender`, `location`, `social_media`, `first_met` and `image`

### CODE REVIEW

**Review by AdamAWiener — DISMISSED** [review]
LGTM! Just a quick question though, is there a specific reason it needs to be `full_name`? Would just `name` work?

**Comment by sdiv877** [comment]
> LGTM! Just a quick question though, is there a specific reason it needs to be `full_name`? Would just `name` work?

The figma design lists the person's name as `full_name` so that's why I went with it. One other thing I just noticed is that there are some extra fields we mustn't have noticed when we made the model initially:
* `gender`
* `location`
* `date_first_met` (instead of `time_added`?)

Should I add these extra fields as well?

**Review by hajineats — APPROVED** [review]
LGTM

**Review by AdamAWiener — APPROVED** [review]
LGTM, good change making the name required too!

File: backend/src/models/person.model.ts [hunk 1]
```diff
@@ -4,24 +4,30 @@
...
-  last_name: string,
+  full_name: string,
   birthday: Date,
+  gender: string,
+  location: string,
+  date_met: Date,
+  how_we_met: string,
   interests: string[],
   organisation: string,
-  time_added: Date,
```
Comment (lin8231) [inline backend/src/models/person.model.ts:12]: How come time_added isn't a thing anymore? I think we need this field to be able to display things on the dashboard for frontend

**Comment by lin8231** [comment]
Btw time_added is unrelated to first_met. First met is a nullable field that the user fills out of when did they first, whereas time_added is the Date.now() timestamp of when the user created that person. It is for displaying the "recently added" things on the dashboard"

**Review by lin8231 — CHANGES_REQUESTED** [review]
Please add time_added back in. See usage here: 
(You might want to name is time_updated)

image

File: backend/src/models/person.model.ts (hunk 1 above)
Comment (sdiv877) [reply backend/src/models/person.model.ts:12]: Got it, have added it back 👍

**Review by lin8231 — APPROVED** [review]
Perfect, thank you!

**Review by hajineats — APPROVED** [review]
LGTM

**Pull request merged**